components/claim/components/ldap_to_email.tsx

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {t} from 'utils/i18n';
+import {isValidPassword, localizeMessage} from 'utils/utils';
+import {ClaimErrors} from 'utils/constants';
+
+import LoginMfa from 'components/login/login_mfa';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+import {PasswordConfig} from '../claim_controller';
+
+import {AuthChangeResponse} from '@mattermost/types/users';
+
+import {SubmitOptions} from './email_to_ldap';
+
+type Props = {
+    email: string | null;
+    switchLdapToEmail: (ldapPassword: string, email: string, password: string, token: string) => Promise<{data?: AuthChangeResponse; error?: {server_error_id: string; message: string}}>;
+    passwordConfig?: PasswordConfig;
+}
+
+const LDAPToEmail = (props: Props) => {
+    const [passwordError, setPasswordError] = useState<string | JSX.Element>('');
```

Comment:
afaik this can be a string type only like the other ones as well ... correct me if i am wrong

Reply:
Nope, `isValidPassword()` might return `JSX.Element` on line 61